src/app/download/download.service.ts

```diff
@@ -9,6 +9,12 @@ export class DownloadService {
     private readonly adAuthoringState: AdAuthoringWorkflowStateContainer
   ) {}
 
+  getAsset() {
+    const assetBase64 = this.adAuthoringState.getValue().fileSrc;
+    const file = this.adAuthoringState.getValue().file;
+    return [assetBase64, file.name];
```

Comment:
Instead of returning an Array, let's create a simple interface to return these values in. It will be more clear when handling the return value elsewhere (i.e. Asset {base64: string, fileName: string}).